mesonbuild/build.py

```diff
@@ -1062,8 +1062,17 @@ def add_deps(self, deps):
     def get_external_deps(self):
         return self.external_deps
 
+    def is_internal(self):
+        if isinstance(self, StaticLibrary) and not self.need_install:
+            return True
+        return False
+
     def link(self, target):
         for t in listify(target, unholder=True):
+            if isinstance(self, StaticLibrary) and t.is_internal():
```

Comment:
This seems to do the isinstance check twice in a row. If so then the method should be called `is_internal_static_lib` and do it only once.

Reply:
No, the first checks if **self** is static, the second checks if **t** is static and not installed.